### PR TITLE
Update 7.0.0

### PR DESCRIPTION
**Pull request by Arishamays1**
# ❄️☆Oblivious 7.0.0 Update ☆❄️
[Jira Ticket](https://anaconda.atlassian.net/browse/PKG-3087)
[Upstream](https://github.com/nthparty/oblivious)
[Dependencies ](https://github.com/nthparty/oblivious/blob/main/pyproject.toml)
# Changes
- Updated version number and `sha256`
- Updated pinnings on dependencies

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
@Arishamays1 To fix a failing test because of this error:
```
    sodium_lib_restore = ristretto.sodium._lib # pylint: disable=protected-access
E   AttributeError: 'NoneType' object has no attribute '_lib'
```
I've added the missing `libsodium` to `host` and `run`. See the reason https://github.com/nthparty/oblivious/blob/03a59af6188b9d30dc8eab7fb0aab7ad483ba266/src/oblivious/ristretto.py#L500-L503

File: recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    - setuptools
+    - wheel
+    - pip
+  run:
+    - python
+    - libsodium >=1.0.18,<1.0.19.0a0
+    - parts >=1.6
+    - fe25519 >=1.4
+    - ge25519 >=1.4
+    - bn254 >=0.1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:30]: Should we set an upper bound?
```suggestion
    - parts >=1.6,<2
    - fe25519 >=1.4,<2
    - ge25519 >=1.4,<2
    - bn254 >=0.1,<0.2
```

**Review by skupr-anaconda — APPROVED** [review]
LGTM but the upper bound

**Comment by Arishamays1** [comment]
> @Arishamays1 To fix a failing test because of this error:
> 
> ```
>     sodium_lib_restore = ristretto.sodium._lib # pylint: disable=protected-access
> E   AttributeError: 'NoneType' object has no attribute '_lib'
> ```
> 
> I've added the missing `libsodium` to `host` and `run`. See the reason https://github.com/nthparty/oblivious/blob/03a59af6188b9d30dc8eab7fb0aab7ad483ba266/src/oblivious/ristretto.py#L500-L503

Thank you, Serhii! I was unsure if we needed to build the sodium library! This definitely clarifies things!

File: abs.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+upload_channels:
+  - sfe1ed40
+
+channels:
+  - sfe1ed40
```
Comment (lorepirri) [inline abs.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    - setuptools
+    - wheel
+    - pip
+  run:
+    - python
+    - libsodium >=1.0.18,<1.0.19.0a0
+    - parts >=1.6,<2
+    - fe25519 >=1.4,<2
+    - ge25519 >=1.4,<2
+    - bn254 >=0.1,<0.2
```
Comment (lorepirri) [inline recipe/meta.yaml]: I think @skupr-anaconda refers to these upper bounds:

```suggestion
    - parts >=1.6
    - fe25519 >=1.4
    - ge25519 >=1.4
    - bn254 >=0.1
```

**Review by lorepirri — APPROVED** [review]
LGTM, but upper bounds, and `channels` not required

**Pull request merged**